src/main/java/uk/ac/ebi/eva/pipeline/io/VepProcess.java

```diff
@@ -0,0 +1,341 @@
+            logger.trace("Finished writing VEP output ({} lines written) to {}", writtenLines, vepOutputPath);
+        });
+        logger.trace("Starting writing VEP output to {}", vepOutputPath);
+        outputCapturer.start();
+    }
+
+    private BufferedReader getBufferedReader(Process process) {
+        return new BufferedReader(new InputStreamReader(process.getInputStream()));
+    }
+
+    private OutputStreamWriter getOutputStreamWriter(String vepOutputPath) throws IOException {
+        return new OutputStreamWriter(new GZIPOutputStream(new FileOutputStream(vepOutputPath, APPEND)));
+    }
+
+    public void write(byte[] bytes) throws IOException {
+        if (!isOpen()) {
+            throw new IllegalStateException("Process must be initialized (hint: call open() before write())");
+        }
+        processStandardInput.write(bytes);
+    }
+
+    public boolean isOpen() {
+        return process != null;
+    }
+
+    public void flush() throws IOException {
+        if (!isOpen()) {
+            throw new IllegalStateException("Process must be initialized (hint: call open() before flush())");
+        }
+        processStandardInput.flush();
+    }
+
+    /**
+     * It is safe to call this method several times; it's idempotent.
+     */
+    public void close() {
+        if (isOpen()) {
+            try {
+                logger.trace("About to close VEP process");
+                flushToPerlStdin();
+                waitUntilProcessEnds(timeoutInSeconds);
+                checkExitStatus();
+                checkOutputWritingStatus();
+            } finally {
+                process = null;
+                logger.trace("VEP process finished");
+            }
+        }
+    }
+
+    private void flushToPerlStdin() {
+        try {
+            processStandardInput.flush();
+            processStandardInput.close();
+        } catch (IOException e) {
+            logger.error("Could not close stream for VEP's stdin", e);
+        }
+    }
+
+    private void waitUntilProcessEnds(Long timeoutInSeconds) {
+        boolean finished;
+        try {
+            boolean processWroteDuringWait;
+            do {
+                long beforeWaiting = System.currentTimeMillis();
+                finished = process.waitFor(timeoutInSeconds, TimeUnit.SECONDS);
+                processWroteDuringWait = beforeWaiting < outputIdleSince.get();
+                if (processWroteDuringWait && !finished) {
+                    logger.debug("Extending the timeout, as the process wrote more lines (it's still active)");
+                }
+            } while (processWroteDuringWait && !finished);
+        } catch (InterruptedException e) {
+            throw new ItemStreamException(e);
+        }
+
+        if (!finished) {
+            String timeoutReachedMessage = "VEP has been idle for more than the timeout (" + timeoutInSeconds
+                    + " seconds). Killed the process.";
+            logger.error(timeoutReachedMessage);
```

Comment:
The message should be logged **after** the process is actually destroyed, otherwise it will be displayed even if it could not be destroyed successfully.